app.go

```diff
@@ -60,7 +61,8 @@ func Loop(args []string, onReady func(*App)) error {
 	defer cerr.free()
 
 	app := App{
-		ready: make(chan struct{}),
+		ready:  make(chan struct{}),
+		server: newServer(http.DefaultServeMux),
```

Comment:
> Hmm won't using DefaultServeMux here make it impossible to use third party http handlers, like gorilla.Mux etc?

Those packages usually implement the `http.Handler` interface, which means they can register with `DefaultServeMux`

> My suggestion would be:
> 
> * export newServer as gallium.NewServer
> * require users to instantiate their own server if they want one
> * require that users pass in their local server url explicitly as in OpenWindow(myserver.BaseURL(), opts)
> 
> That will also mean that users will have the flexibility to avoid opening any network ports at all if they don't want a server, or to have multiple servers if they want that (e.g. one per window). What do you think?

If we go with the approach in PR #19, we could do the following:

* By default, start the HTTP server using `http.DefaultServeMux`
* Allow apps to implement the following interface `type HandlerHTTP interface { HandlerHTTP() http.Handler }`  to set a custom server to use or disable the local server entirely.
* `OpenWindow`, when called with a relative URL, would load the path in the local server with the authentication parameter.

## Using DefaultServeMux by default


```go
package webapp

func init() {
    HandleFunc("/", handler)
}

type webapp struct {}

func New() *webapp {
    return &webapp{}
}

func (*webapp) Start(app *gallium.App) {
    app.OpenWindow("/", opts)
}
```

## Using custom server

```go
package webapp

type webapp struct {
    server *ServeMux
}

func New() *webapp {
    s := NewServeMux()
    s.HandleFunc("/", handler)
    return &webapp{server: s}
}

func (*webapp) Start(app *gallium.App) {
    app.OpenWindow("/", opts)
}

func (w *webapp) HandlerHTTP() http.Handler {
    return w.server
}
```

## Disabling the HTTP Server

```go
package webapp

type webapp struct {}

func New() *webapp {
    return &webapp{}
}

func (*webapp) Start(app *gallium.App) {
    app.OpenWindow("http://example.com", opts)
}

func (w *webapp) HandlerHTTP() http.Handler {
    return nil
}
```